applications/GeoMechanicsApplication/tests/cpp_tests/custom_geometries/test_interface_geometry.cpp

```diff
@@ -0,0 +1,367 @@
+{
+    const auto geometry = CreateFourNodedLineInterfaceGeometry();
+    const auto xi       = array_1d<double, 3>{0.5, 0.0, 0.0};
+
+    KRATOS_EXPECT_RELATIVE_NEAR(geometry.DeterminantOfJacobian(xi), 3.25, 1e-6)
+}
+
+KRATOS_TEST_CASE_IN_SUITE(InterfaceGeometry_ReturnsCorrectDeterminantOfJacobian_ForSixNodedGeometry,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    const auto geometry = CreateSixNodedLineInterfaceGeometry();
+    const auto xi       = array_1d<double, 3>{0.5, 0.0, 0.0};
+
+    KRATOS_EXPECT_RELATIVE_NEAR(geometry.DeterminantOfJacobian(xi), std::sqrt(9.01), 1e-6)
+}
+
+KRATOS_TEST_CASE_IN_SUITE(InterfaceGeometry_Throws_WhenCallingInverseJacobian, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    const auto geometry = CreateFourNodedLineInterfaceGeometry();
+    const auto xi       = array_1d<double, 3>{0.5, 0.0, 0.0};
+
+    Matrix result;
+    KRATOS_EXPECT_EXCEPTION_IS_THROWN(
+        geometry.InverseOfJacobian(result, xi),
+        "Inverse of Jacobian is not implemented for the line interface geometry")
+}
+
+KRATOS_TEST_CASE_IN_SUITE(FourNodedLineInterfaceGeometry_LengthReturnsTheLengthOfUnderlyingLineGeometry,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    const auto geometry = CreateFourNodedLineInterfaceGeometry();
+
+    constexpr auto expected_length = 6.5;
+    KRATOS_EXPECT_RELATIVE_NEAR(geometry.Length(), expected_length, 1e-6)
+}
+
+KRATOS_TEST_CASE_IN_SUITE(SixNodedLineInterfaceGeometry_LengthReturnsTheLengthOfUnderlyingLineGeometry,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    const auto geometry = CreateSixNodedLineInterfaceGeometry();
+
+    constexpr auto expected_length = 6.504159;
+    KRATOS_EXPECT_RELATIVE_NEAR(geometry.Length(), expected_length, 1e-6)
+}
+
+KRATOS_TEST_CASE_IN_SUITE(FourNodedLineInterfaceGeometry_DomainSizeReturnsTheLengthOfUnderlyingLineGeometry,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    const auto geometry = CreateFourNodedLineInterfaceGeometry();
+
+    constexpr auto expected_length = 6.5;
+    KRATOS_EXPECT_RELATIVE_NEAR(geometry.DomainSize(), expected_length, 1e-6)
+}
+
+KRATOS_TEST_CASE_IN_SUITE(SixNodedLineInterfaceGeometry_DomainSizeReturnsTheLengthOfUnderlyingLineGeometry,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    const auto geometry = CreateSixNodedLineInterfaceGeometry();
+
+    constexpr auto expected_length = 6.504159;
+    KRATOS_EXPECT_RELATIVE_NEAR(geometry.DomainSize(), expected_length, 1e-6)
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GlobalCoordinatesAreCorrectlyMappedToLocalCoordinateForFourNodedLineInterfaceGeometry,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    const auto geometry = CreateFourNodedLineInterfaceGeometry();
+
+    auto       result          = array_1d<double, 3>{};
+    const auto global_position = array_1d<double, 3>{4.375, 0.1, 0.0};
```

Comment:
Seems to be engineered such that it is on the midline. What happens if it is not ( or even outside the interface ) ? ( That a point lies outside the interface element is not necessarily wrong as it just a change of coordinate system. Use of local coordinates outside the interface element may be wrong, but that depends on the purpose.